### ISSUE TITLE
Enable signing of encrypted messages

### ISSUE THREAD

**Issue by lbort** [mailvelope/mailvelope#472]
I just installed mailvelope, up to now I was only using thunderbird+enigmail. I sign every email I send, and encrypt+sign the ones to recipients that I have a key of. I think from a security perspective, every encrypted Email should be signed, because only the signature ensures that the message was not changed on the way, or forged altogether. Is this actively developped right now?

**Comment by JakeChampion** [comment]
Mailvelope can sign and encrypt a message, yes.

**Comment by toberndo** [comment]
Signing is now available:

![bildschirmfoto 2017-01-19 um 11 48 14](https://cloud.githubusercontent.com/assets/1164474/22103916/42a37078-de3d-11e6-82bf-50c63486e9e1.png)

**Comment by lbort** [comment]
Hi and thanks for the quick reply, 
I read that it should be possible in the meantime, but do not find a way to get it to work. somehow I am missing these options, as shown attached. 

Does that depend on the provider I use (which is posteo.de)? 
Or is it a problem that I created the key with another program and imported it?
Or (most likely) is this related to Issue #345  since I have different subkeys? If so, maybe mark this as a duplicate of #345. 
![options_mailvelope](https://cloud.githubusercontent.com/assets/20089789/22115466/622e3ca4-de6d-11e6-8c51-7aa496fbffe7.png)

**Comment by lbort** [comment]
UPDATE: 
I tried a few different things, and am quite confused about what I should rename this ticket, but there is definitively an issue. I am still not sure if my rename of the ticket is exactely right... 

I figured that there is two options, having the API switch on, or off. 

**If the API is ON:**
1.) Encrypt and Sign *does not* work if the API is used, at least for posteo.de
2.) Incoming PGP/MIME messages which were encrypted and signed can be read, and it displays that the signature is valid
3.) Incoming PGP/MIME messages which are only signed can be read, but the attached signature.asc is not evaluated, but shown as an attachment instead. 

**If the API is OFF:** 
1.) Encrypt and sign of outgoing messages *does* work, and the window looks just like the screenshot of toberndo
2.) Incoming PGP/MIME messages which were encrypted and signed can be read, but it does not verify the signature
3.) Incoming PGP/MIME messages which were only signed are not verified. 

I think there are more possibilities and combinations, but this is how far I got now. If someone is interested in a more complete list, I have thunderbird with enigmail, k9-mail and mailvelope available, and could test sending/recieving with PGP/MIME or PGP/INLINE in all directions. But I will do so only if it is helpful to actually fix the problem. 

Basically, I think in the API-ON-case it should be possible to decrypt any incoming message, be it INLINE or MIME, and validate the signature if there is one. And for outgoing messages, the API-ON-case should have the same functionality as the API-off case. 

Relevant other issues I found so far: #457 (which I do not understand. verifying a signature is not dependent on the provider, or is it?)

**Comment by toberndo** [comment]
> If the API is ON:
> 1.) Encrypt and Sign does not work if the API is used, at least for posteo.de

This is supported in the API, but the mail provider has to turn on this flag. Not sure if Posteo has a setting for that?

> 3.) Incoming PGP/MIME messages which are only signed can be read, but the attached > signature.asc is not evaluated, but shown as an attachment instead.

sign-only PGP/MIME is currently not supported in the API. Neither sending nor verifying such messages.

> If the API is OFF:
> 2.) Incoming PGP/MIME messages which were encrypted and signed can be read, but it does 
> not verify the signature

This currently only works for the providers Gmail, Yahoo and Outlook.com as we require the sender address for signature verification and that's only implemented for these providers.

3.) Incoming PGP/MIME messages which were only signed are not verified.

Same as with the API case, no support here at the moment.
